annotations/certmanager-annotations/src/main/java/io/dekorate/certmanager/decorator/AddCertificateResourceDecorator.java

```diff
@@ -26,26 +25,25 @@
 import io.fabric8.certmanager.api.model.v1.X509Subject;
 import io.fabric8.certmanager.api.model.v1.X509SubjectBuilder;
 import io.fabric8.kubernetes.api.model.Duration;
-import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
 
 public class AddCertificateResourceDecorator extends ResourceProvidingDecorator<KubernetesListBuilder> {
 
+  private final String name;
   private final CertificateConfig config;
 
-  public AddCertificateResourceDecorator(CertificateConfig config) {
+  public AddCertificateResourceDecorator(String name, CertificateConfig config) {
+    this.name = name;
     this.config = config;
   }
 
   @Override
   public void visit(KubernetesListBuilder list) {
-    HasMetadata meta = getMandatoryDeploymentHasMetadata(list, ANY);
     CertificateBuilder builder = new CertificateBuilder();
 
     // metadata
     builder.withNewMetadata()
-        .withName(getName(meta))
-        .withNamespace(meta.getMetadata().getNamespace())
```

Comment:
Updating the namespace was not necessary as there is a decorator that populates all the resources with the correct namespace.